src/compact/compact.test.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable unicorn/no-null */
+import test from "tape"
+
+import { compact, compactMany } from "./compact"
+
+const lambda = () => null
+
+test("compact", t => {
```

Comment:
One test that is missing and would of exposed the `typeof` error is testing for `null`.

Either add it to the `compactMany([{}, 2, null, undefined])` that will proxy to the core `_compact` or add an explicit `compact(null)`.

I should write at some point a DoD I guess, to always test `null` and `undefined`, `*Many` and `*With` patterns, etc